guides/Writing API Documentation.md

```diff
@@ -0,0 +1,149 @@
+
+## Grammar
+
+This starts with objectively correct writing. Developers reading the Expo docs shouldn't find obvious mistakes and come away feeling less confident in Expo. Write with correct grammar, punctuation, and spelling. These are a few grammar-related points that we historically could have improved:
+
+- Use double quotes in prose.
+  - Write:
+    > Set the field named "id" to your project's ID.
+  - Don't write:
+    > Set the field named 'id' to your project's ID.
+- Use em dashes (—), semicolons, or colons rather than commas to split phrases that read as separate sentences.
+  - Write:
+    > JavaScript has come a long way; it was originally written in 10 days.
+  - Don't write:
+    > JavaScript has come a long way, it was originally written in 10 days.
+- Use the Oxford comma.
+
+These aren't the only points to consider; we're looking to be fully grammatically correct.
+
+## Spelling and wording
+
+We use American English like other relevant documentation including the W3C and WHATWG HTML specifications, Apple's iOS documentation, and Google's Android documentation.
+
+Some words have multiple legitimate ways to spell them. One of the most notorious examples in the Expo codebase is "canceled" vs. "cancelled". We tend to write "cancellation" instead of "cancelation" but are undecided between "canceled" and "cancelled". (This isn't just a problem for Expo; even the TC39 proposal for JS cancellation tokens is inconsistent!) Prefer "canceled" and "canceling".
+
+Write out words instead of abbreviating them, unless the abbreviations are ubiquitous. For example, write out "parameter" and "configuration" instead of using "param" and "config", and use abbreviations like "HTML", "API", "laser", and "mutex".
+
+Succinctness > verbosity > ambiguity.
+
+Properly capitalize "GitHub" and "JavaScript".
+
+## Technical accuracy and precision
+
+Writing our documentation to be accurate and precise helps developers use our APIs correctly. This is especially true when developers are knowledgeable about the topics at hand and inaccurate documentation could mislead them.
+
+Teaching or explaining a concept usually requires understanding it well. We need to have clarity of mind and a correct understanding of various programming concepts to write good documentation.
+
+These are some topics that often come up with Expo development:
+
+### Concurrency and parallelism
+  - Concurrency describes two tasks logically running together. Two concurrent tasks may each start before the other finishes.
+  - Parallelism describes two tasks physically running at the same time. Two network requests or processes running on two CPU cores are examples of parallelism.
+  - It is possible, and very common with JavaScript, to have concurrency without parallelism. Two async functions without external I/O will run concurrently but not in parallel because the JavaScript microtask scheduler will interleave the async functions but run them single-threadedly.
```

Comment:
newline after heading and remove indent before - maybe? same for other sections below